pkg/release/release.go

```diff
@@ -379,10 +402,12 @@ func (r *Release) install(client helm.Client, hr *v1.HelmRelease, chart chart, v
 		Wait:       hr.Spec.Wait,
 	})
 	if err != nil {
+		status.SetInstallPhase(r.hrClient.HelmReleases(hr.Namespace), hr, v1.HelmReleasePhaseFailed)
 		status.SetStatusPhase(r.hrClient.HelmReleases(hr.Namespace), hr, v1.HelmReleasePhaseFailed)
 		err = fmt.Errorf("installation failed: %w", err)
 		return
 	}
+	status.SetInstallPhase(r.hrClient.HelmReleases(hr.Namespace), hr, v1.HelmReleasePhaseSucceeded)
```

Comment:
This is technically also a duplicate of the call below, except that it does not push the revision. I think this can be prevented by resetting the counter on a `v1.HelmReleasePhaseSucceeded` as is already done for successful upgrades after a rollback.

Reply:
I am not sure I follow, failed will increment and success/install reset?

Reply:
`SetInstallPhase` and `SetStatusPhaseWithRevision` do the same thing: they update the status of the HelmRelease with the given phase, except the install version takes the reset and increment of the counter into account.

I think this reset logic can be moved to `SetCondition` so that `SetInstallPhase` no longer needs to be called here as no additional context is required for the reset (but only for the increment on failure).